gossip/client_test.go

```diff
@@ -138,26 +138,32 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
```

Comment:
Maybe we should increase that timeout globally instead of introducing another retry mechanism (see #5218)

We use `SucceedsSoon` for several reasons: better error reporting (panics in tests lose information), smaller timeout scoping (the command line timeout is per package instead of per test, so you have to manually change it to iterate on one test that's timing out), and the exponential backoff takes the guesswork out of choosing a sleep time.